extensions-core/hdfs-storage/src/main/java/org/apache/druid/storage/hdfs/HdfsStorageDruidModule.java

```diff
@@ -91,7 +92,15 @@ public void configure(Binder binder)
     ClassLoader currCtxCl = Thread.currentThread().getContextClassLoader();
     try {
       Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
-      FileSystem.get(conf);
+      FileSystem fs = FileSystem.get(conf);
+
+      // If cache is enabled, need to check the FileSystem object by access hdfs because FileSystem object will be used later, like in push stage.
+      // If FileSystem is invalid, the peon task should stop immediately.
+      // See https://github.com/apache/druid/pull/14276
+      boolean disableCache = conf.getBoolean("fs.hdfs.impl.disable.cache", false);
```

Comment:
This is going to do the verification at binding time. IMO, ideally it would happen at start time, so we can separate binding errors from runtime errors. Would it work to move this validation into a start handler?

You can use a technique similar to `HdfsStorageAuthentication`, which does kerberos authentication at start time.

Reply:
Thanks for your suggestion. I add class `HdfsStorageAvailabilityChecker`.